### PR TITLE
Feature/149 search UI enhancements

### PR DESCRIPTION
**Pull request by davideastmond**
- Feel free to add the included command in your `launch.json` in vscode to debug TS from the launch.sample.json
- Also updates to `tsconfig` are included to assist with TS debugging in vscode
- added "mingo" package - to help with querying objects using a mongo-like query structure.
- The sandbox code included is to help test mingo queries

- Updates to the search field in the nav bar.
- Local search searches connections, threads and threadComment data from the local store.
- Local connection results will link to the user profile. Links to threads/Comments will happen when threads page is done.

### CODE REVIEW

File: app/src/services/search/local/local-search.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import mingo from "mingo";
+import assert from "assert";
+
+export type ConnectionsSearchMatch = {
```
Comment (lily-law) [inline app/src/services/search/local/local-search.ts]: Just curious, why types instead of interface?

**Review by lily-law — APPROVED** [review]
:cool: looking good so far :+1:

File: app/src/services/search/local/local-search.ts (hunk 1 above)
Comment (davideastmond) [reply app/src/services/search/local/local-search.ts]: I figured since we're not extending the type we could use types instead.

Comment (davideastmond) [reply app/src/services/search/local/local-search.ts]: On second thought, I think interfaces are a better fit here.

File: app/src/components/search/Search.tsx [hunk 2]
```diff
@@ -1,34 +1,81 @@
...
 } from "../../pages/search/searchSlice";
 import "./Search.css";
+import { selectUserConnections } from "../../pages/profile/profileSlice";
+import { selectThreads } from "../../pages/home/homeSlice";
+import { LocalSearchResults } from "./sub-components/local-search-results";
 
 function Search({ className }: { className?: string }) {
   const query = useSelector(selectSearchQuery, shallowEqual);
   const dispatch = useDispatch();
+  const [cancelQueryVisible, setCancelQueryVisible] = useState<boolean>(false);
```
Comment (lily-law) [inline app/src/components/search/Search.tsx]:
```suggestion
  const [cancelQueryVisible, setCancelQueryVisible] = useState<boolean>(query.length > 0);
```

**Review by lily-law — APPROVED** [review]
Nice this is looking great, really nice with the drop down functionality 👍 
Just the one suggestion, basically when you change page and there's still a query the X doesn't show

**Pull request merged**